airbyte-webapp/src/components/connectorBuilder/StreamTestingPanel/ConfigMenu.tsx

```diff
@@ -88,14 +84,17 @@ export const ConfigMenu: React.FC<ConfigMenuProps> = ({ className }) => {
                   bodyClassName={styles.formContent}
                   footerClassName={styles.inputFormModalFooter}
                   selectedConnectorDefinitionSpecification={jsonManifest.spec}
-                  formValues={formValues}
+                  formValues={configJson}
                   onSubmit={async (values) => {
-                    setConfigString(JSON.stringify(values.connectionConfiguration, null, 2) ?? "");
+                    setConfigJson(values);
                     setIsOpen(false);
                   }}
                   onCancel={() => {
                     setIsOpen(false);
                   }}
+                  onReset={() => {
+                    setConfigJson({});
```

Comment:
Slightly weird UX here: If I open the config menu and all the inputs are blank, then I type a value into one of them, then click Reset, it doesn't clear the value. 
It seems like it only clears values if they had been saved to the form before, likely because values that have not yet been saved arent set on ConfigJson so setting that to an empty object doesn't affect unsaved values.

Ideally this would always clear all values in all inputs when clicked.

Reply:
Fixed by also calling resetForm on the formik context to purge local formik state.